### PR TITLE
Zbs253/gauss fit

### PR DESCRIPTION
**Pull request by zack-erisyon**
Gauss 2D fitter

You can ignore all the files in levmar-2-6 folder (the bulk of this PR) as they are just a vendoring of an open-source LevMar code in C.

Unit and integration tests are passing

### CODE REVIEW

File: plaster/run/sigproc_v2/c/gauss2_fitter.c [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+#include "stdlib.h"
+#include "math.h"
+#include "float.h"
+#include "time.h"
+#include "string.h"
+#include "levmar.h"
+#include "c_common.h"
+#include "gauss2_fitter.h"
+#ifndef LM_DBL_PREC
+#error Demo program assumes that levmar has been compiled with double precision, see LM_DBL_PREC!
```
Comment (danwinkler) [inline plaster/run/sigproc_v2/c/gauss2_fitter.c:11]: Demo program? I'm guessing this was copied over from some levmar example?

File: plaster/run/sigproc_v2/c/gauss2_fitter.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    return lib
+
+
+# TODO: Mo ve this and NNV2Exception to a generic CException
+class Gauss2FitException(Exception):
+    def __init__(self, s):
+        super().__init__(s.decode("ascii"))
+
+
+class Gauss2FitParams:
```
Comment (danwinkler) [inline plaster/run/sigproc_v2/c/gauss2_fitter.py:80]: Should this be an `IntEnum`?

**Review by danwinkler — APPROVED** [review]

File: plaster/run/sigproc_v2/c/gauss2_fitter.c (hunk 1 above)
Comment (zack-erisyon) [reply plaster/run/sigproc_v2/c/gauss2_fitter.c:11]: Correct. Changed comment

File: plaster/run/sigproc_v2/c/gauss2_fitter.py (hunk 2 above)
Comment (zack-erisyon) [reply plaster/run/sigproc_v2/c/gauss2_fitter.py:80]: Good call. Changed

**Pull request merged**